Simplified/NYPLSettingsAccountDetailViewController.m

```diff
@@ -207,7 +207,7 @@ - (void)viewDidLoad
     [self.view addSubview:activityIndicator];
     [activityIndicator startAnimating];
     self.loading = true;
-    [self.selectedAccount loadAuthenticationDocumentWithCompletion:^(BOOL success) {
+    [self.selectedAccount loadAuthenticationDocumentWithUserAccountProvider:self completion:^(BOOL success) {
```

Comment:
here you could also pass the businessLogic and make that the implementor of `NYPLUserAccountProvider` instead.